content/components/data-table.mdx

```diff
@@ -535,6 +543,37 @@ match that query.
 />
 </Box>
 
+### Row selection
+
+{/* ![Video of DataTable with row selection. The first 3 rows are selected, and the select-all checkbox is in an indeterminate state. The header changes to show the batch row actions when one or more rows are selected.]() */}
+
+A data table component can support the ability to have one or more rows selected. A batch of selected rows can then have actions applied to them.
+
+Selection functionality must be present on **all** rows of a data table. Only having a subset of rows be selectable **is not** supported.
+
+Each row may be selected or deselected individually, or all at once by using the "Select all rows" checkbox in the header of the checkbox column.
+
+### Selecting all rows
+
+{/* ![Video of selecting all rows w/ the "Select all rows" checkbox, deselecting the first three rows, then deselecting all rows w/ the "Select all rows" checkbox]() */}
+
+Activating the "Select all rows" checkbox in the header of the checkbox column will place **all** data table body rows on the currently displayed page in a selected state.
+
+The "Select all rows" checkbox is also placed in a checked state if **all** data table rows are selected on the currently displayed paginated page.
+
+If one or more data table rows are placed in a selected state, The "Select all rows" checkbox `input` is also placed in an [indeterminate state](https://html.spec.whatwg.org/multipage/input.html#checkbox-state-(type=checkbox)).
+
+Un-checking the "Select all rows" checkbox will place all associated data table rows in an unchecked state.
+
+### Row selection considerations for pagination
+
+{/* ![Video of selecting all rows on page one, then going to page two, then going back to page one.]() */}
+
+If the data table is paginated, the "Select all rows" checkbox is activated, all rows on the currently displayed page are selected. Rows on pages that are not the currently displayed will **not** be selected. 
+For example, consider a paginated data table that displays 25 rows per page, with a total of 400 rows in the table. All rows on the current page are selected by the user. Deleting the selected rows should remove only the displayed rows, as to avoid unintentionally applying a destructive act to the  other 375 rows.
```

Comment:
```suggestion
If the DataTable is paginated, the "Select all rows" checkbox is activated, all rows on the currently displayed page are selected. Rows on pages that are not the currently displayed will **not** be selected. 

For example, consider a paginated DataTable that displays 25 rows per page, with a total of 400 rows in the table. All rows on the current page are selected by the user. Deleting the selected rows should remove only the displayed rows, as to avoid unintentionally applying a destructive act to the other 375 rows.
```